File: 3sem/4lab/Banks.Console/Handlers/EmptyHandler.cs
using Banks.Services;

namespace Banks.Console.Handlers;

public class EmptyHandler : Handler
{
    public override void Handle(string request, CentralBank centralBank)
    {
        ArgumentNullException.ThrowIfNull(request);
        ArgumentNullException.ThrowIfNull(centralBank);

        NextHandler?.Handle(request, centralBank);
    }
}

File: 4sem/1lab/src/main/java/banksRemake/presentation/handlers/EmptyHandler.java
package banksRemake.presentation.handlers;

import banksRemake.domain.exceptions.InvalidBankAccountOperationException;
import banksRemake.domain.exceptions.InvalidBankOperationException;
import banksRemake.domain.exceptions.InvalidDepositInterestRateSelectorException;
import banksRemake.domain.exceptions.InvalidMoneyGapException;
import banksRemake.domain.services.CentralBank;

import java.util.Scanner;

public class EmptyHandler extends HandlerBase {
    @Override
    public void handle(String request, CentralBank centralBank, Scanner scanner) throws InvalidBankOperationException, InvalidBankAccountOperationException, InvalidDepositInterestRateSelectorException, InvalidMoneyGapException {
        if (getNext() != null) {
            getNext().handle(request, centralBank, scanner);
        }
    }
}
